File: Backend/Frontend/src/app/models/BesoinCommercial/besoin-commercial.ts
import { Entreprise } from "../Entreprise/entreprise";

export class BesoinCommercial {
    idBesoin!: number;
    titre!: string;
    description!: string;
    entreprise!: Entreprise;
    besoinetat!:string;
  }
  export enum Besoinetat {
    Open , Inprogress, Closed
  }

File: Backend/Relations/src/main/java/tn/esprit/Entities/BesoinCommercial.java
package tn.esprit.Entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class BesoinCommercial {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idBesoin ;
    private String titre ;
    private String description ;
    @Enumerated(EnumType.STRING)
    private Besoinetat besoinetat ;
    @ManyToOne
    private Entreprise entreprise ;

}
